benchmark/compare.js

```diff
@@ -1,181 +1,73 @@
 'use strict';
-var usage = 'node benchmark/compare.js ' +
-            '<node-binary1> <node-binary2> ' +
-            '[--html] [--red|-r] [--green|-g] ' +
-            '[-- <type> [testFilter]]';
 
-var show = 'both';
-var nodes = [];
-var html = false;
-var benchmarks;
-
-for (var i = 2; i < process.argv.length; i++) {
-  var arg = process.argv[i];
-  switch (arg) {
-    case '--red': case '-r':
-      show = show === 'green' ? 'both' : 'red';
-      break;
-    case '--green': case '-g':
-      show = show === 'red' ? 'both' : 'green';
-      break;
-    case '--html':
-      html = true;
-      break;
-    case '-h': case '-?': case '--help':
-      console.log(usage);
-      process.exit(0);
-      break;
-    case '--':
-      benchmarks = [];
-      break;
-    default:
-      if (Array.isArray(benchmarks))
-        benchmarks.push(arg);
-      else
-        nodes.push(arg);
-      break;
-  }
+const fork = require('child_process').fork;
+const path = require('path');
+const CLI = require('./_cli.js');
+
+//
+// Parse arguments
+//
+const cli = CLI(`usage: ./node benchmark/compare.js <type> ...
```

Comment:
Can you elaborate on that note, I think it is very specific, this is the message you get now.

```
usage: ./node benchmark/compare.js <type> ...
  Run each benchmark in the <type> directory many times using two diffrent
  node versions. More than one <type> directory can be specified. The output is
  formatted as csv, which can be processed using for example 'compare.R'.

  --new    ./new-node-binary  new node binary (required)
  --old    ./old-node-binary  old node binary (required)
  --runs   30                 number of samples
  --filter pattern            string to filter benchmark scripts
  --set    variable=value     set benchmark variable (can be repeated)
```

I choose this order, because it could be implemented using less code.

I will try and change the argument order, this appears to cause a lot of confusion for many people, but I would love to understand why.